### PR TITLE
WFLY-14385 Ensure wildfly-clustering-server ProtoStream marshallers use valid protobuf encoding

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-14385

Requires: https://github.com/wildfly/wildfly/pull/13951

### CODE REVIEW

**Comment by pferraro** [comment]
@rhusar Please review

File: clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/ChannelCommandDispatcherFactory.java [hunk 1]
```diff
@@ -232,6 +232,7 @@ public Group getGroup() {
     @Override
     public <C> CommandDispatcher<C> createCommandDispatcher(Object id, C commandContext, ClassLoader loader) {
         ByteBufferMarshaller dispatcherMarshaller = this.createMarshaller(loader);
+        System.out.println(id + " dispatcher will use " + dispatcherMarshaller.getClass().getName());
```
Comment (rhusar) [inline clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/ChannelCommandDispatcherFactory.java:235]: Please rid of this System.out.

File: clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/CommandResponseMarshaller.java [hunk 2]
```diff
@@ -47,19 +47,29 @@
...
+        try {
         ByteBuffer buffer = this.marshaller.write(object);
 
         int length = buffer.limit() - buffer.arrayOffset();
         IndexSerializer.VARIABLE.writeInt(stream, length);
         stream.write(buffer.array(), buffer.arrayOffset(), length);
+        } catch (IOException | RuntimeException | Error e) {
+            e.printStackTrace(System.out);
+            throw e;
+        }
```
Comment (rhusar) [inline clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/CommandResponseMarshaller.java:59]: Please run a formatter to fix indentation.

File: clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/CommandResponseMarshaller.java [hunk 3]
```diff
@@ -47,19 +47,29 @@
...
+        try {
         int size = IndexSerializer.VARIABLE.readInt(stream);
         byte[] bytes = new byte[size];
         stream.readFully(bytes);
         ByteBuffer buffer = ByteBuffer.wrap(bytes);
         return this.factory.isUnknownForkResponse(buffer) ? NoSuchService.INSTANCE : this.marshaller.read(buffer);
+        } catch (IOException | RuntimeException | Error e) {
+            e.printStackTrace(System.out);
+            throw e;
+        }
```
Comment (rhusar) [inline clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/CommandResponseMarshaller.java:73]: Please run a formatter to fix indentation.

**Review by rhusar — CHANGES_REQUESTED** [review]

**Review by rhusar — APPROVED** [review]
👍

**Pull request merged**